src/storage/mutate/AddEdgesProcessor.cpp

```diff
@@ -153,6 +153,24 @@ void AddEdgesProcessor::doProcessWithIndex(const cpp2::AddEdgesRequest& req) {
         visited.reserve(newEdges.size());
         for (auto& newEdge : newEdges) {
             auto edgeKey = *newEdge.key_ref();
+            auto l = std::make_tuple(spaceId_,
+                                     partId,
+                                     (*edgeKey.src_ref()).getStr(),
+                                     *edgeKey.edge_type_ref(),
+                                     *edgeKey.ranking_ref(),
+                                     (*edgeKey.dst_ref()).getStr());
+            if (std::find(dummyLock.begin(), dummyLock.end(), l) == dummyLock.end()) {
+                if (!env_->edgesML_->try_lock(l)) {
+                    LOG(ERROR) << folly::format("edge locked : src {}, type {}, rank {}, dst {}",
+                                                (*edgeKey.src_ref()).getStr(),
+                                                *edgeKey.edge_type_ref(),
+                                                *edgeKey.ranking_ref(),
+                                                (*edgeKey.dst_ref()).getStr());
+                    code = nebula::cpp2::ErrorCode::E_DATA_CONFLICT_ERROR;
+                    break;
+                }
+            }
+            dummyLock.emplace_back(std::move(l));
```

Comment:
the records may be duplicated, maybe you move 173 lines to 170 lines below

Reply:
> the records may be duplicated, maybe you move 173 lines to 171 lines below

Aha，industrious panda. I think repetition is allowed here, and param `dedup = true` of `MemoryLockGuard` constructor method (line 303). WDYT?

Reply:
> the records may be duplicated, maybe you move 173 lines to 171 lines below

You are right. your way is safer.

Reply:
> > the records may be duplicated, maybe you move 173 lines to 171 lines below
> 
> You are right. your way is safer.

thanks, fixed.